rules_gapic/bazel/src/main/java/com/google/api/codegen/bazel/ApiVersionedDir.java

```diff
@@ -0,0 +1,212 @@
+    return serviceConfigJsonPath;
+  }
+
+  String getName() {
+    return name;
+  }
+
+  String getAssemblyName() {
+    return assemblyName;
+  }
+
+  String getVersion() {
+    return version;
+  }
+
+  void parseYamlFile(String fileName, String fileBody) {
+    // It is a gapic yaml
+    Matcher m = GAPIC_YAML_TYPE.matcher(fileBody);
+    if (m.find()) {
+      gapicYamlPath = fileName;
+
+      m = LANG_PACKAGES.matcher(fileBody);
+      while (m.find()) {
+        String lang = m.group("lang");
+        langGapicPackages.put(lang, m.group("package"));
+        String interfaceNames = m.group("interfaceNames");
+        Map<String, String> interfaceSubs = new HashMap<>();
+        if (interfaceNames != null) {
+          Matcher subM = INTERFACE_NAMES.matcher(interfaceNames);
+          while (subM.find()) {
+            interfaceSubs.put(subM.group("name"), subM.group("val"));
+          }
+        }
+        langGapicNameOverrides.put(lang, interfaceSubs);
+      }
+
+      return;
+    }
+
+    // It is a service yaml
+    m = SERVICE_YAML_TYPE.matcher(fileBody);
+    if (m.find()) {
+      serviceYamlPath = fileName;
+    }
+  }
+
+  void parseProtoFile(String fileName, String fileBody) {
+    protos.add(fileName);
+
+    // Parse proto package
+    Matcher m = PROTO_PACKAGE.matcher(fileBody);
+    if (m.find() && (protoPackage == null || name == null || version == null)) {
+      protoPackage = m.group("protoPackage");
+      String[] tokens = protoPackage.split("\\.");
+      if (tokens.length >= 1 && name == null) {
+        name = tokens[tokens.length - 1];
+        assemblyName = name;
+      }
+      if (tokens.length >= 2) {
+        String ver = tokens[tokens.length - 1];
+        if (ver.length() >= 2 && ver.startsWith("v") && Character.isDigit(ver.charAt(1))) {
+          version = ver;
+          name = tokens[tokens.length - 2];
+          assemblyName = name;
+          if (tokens.length >= 3) {
+            String topPackage = tokens[tokens.length - 3];
+            if (!"google".equals(topPackage) && !"cloud".equals(topPackage)) {
+              assemblyName = topPackage + '-' + name;
+            }
+          }
+        }
+      }
+    }
+
+    // Parse imports
+    m = IMPORTS.matcher(fileBody);
+    while (m.find()) {
+      imports.add(m.group("import"));
+    }
+
+    // Parse file-level options
+    m = PROTO_OPTIONS.matcher(fileBody);
+    while (m.find()) {
+      String optName = m.group("optName");
+      String optValue = m.group("optValue");
+      optName = optName.split("_")[0];
+      if (!langProtoPackages.containsKey(optName)) {
+        langProtoPackages.put(optName, optValue);
+      }
+    }
+
+    // Parse declared service names
+    m = SERVICE.matcher(fileBody);
+    while (m.find()) {
+      services.add(m.group("service"));
+    }
+  }
+
+  void parseJsonFile(String fileName, String fileBody) {
+    if (fileBody.contains("methodConfig") && fileBody.contains("retryPolicy")) {
```

Comment:
Ok. This wasn't used yet (since we decided to not include grpc_service_configs yet). I left it to check only for the methodConfig portion (I guess it should be good enough).